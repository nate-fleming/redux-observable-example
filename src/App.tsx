import React from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import logo from "./logo.svg";
import "./App.css";
import table from "./img/table.png";
import paddle from "./img/paddle.png";

import { pingAction, pongAction, togglePlayingAction } from "./actions/actions";

interface IProps {
  isPing: boolean;
  isPong: boolean;
  isPlaying: boolean;
  togglePlayingAction: any;
  pingAction: any;
}

const App: React.FC<IProps> = ({
  isPing,
  isPong,
  isPlaying,
  togglePlayingAction,
  pingAction
}) => {
  console.log(isPlaying, isPing, isPong);
  return (
    <div className="App">
      <header className="App-header">
        <img src={table} className="Table" alt="logo" />
        <button
          onClick={() => {
            togglePlayingAction();
            pingAction();
            console.log("clicked");
          }}
          className="Start-Button"
        >
          {isPlaying ? "Stop" : "Start"}
        </button>
        <div style={{ display: "flex", width: "100%" }}>
          {isPlaying && (
            <>
              {isPing && (
                <div className="Ping">
                  Ping
                  <img src={paddle} className="Left-Paddle" alt="paddle" />
                </div>
              )}
              {isPong && (
                <div className="Pong">
                  Pong
                  <img src={paddle} className="Left-Paddle" alt="paddle" />
                </div>
              )}
            </>
          )}
        </div>
      </header>
    </div>
  );
};

const mapStateToProps = (state: any) => {
  return {
    isPing: state.isPing,
    isPong: state.isPong,
    isPlaying: state.isPlaying
  };
};

const matchDispatchToprops = (dispatch: any) => {
  return bindActionCreators(
    { pingAction, pongAction, togglePlayingAction },
    dispatch
  );
};

export default connect(mapStateToProps, matchDispatchToprops)(App);
