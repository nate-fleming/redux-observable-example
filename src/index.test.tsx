import React from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import { render } from "@testing-library/react";
import reducer from "../src/reducers/reducers";

import App from "./App";

const renderWithRedux = (
  reducer: any,
  initialState: any,
  children: JSX.Element
) => {
  const mockStore = createStore(reducer, initialState);
  return render(<Provider store={mockStore}>{children}</Provider>);
};

describe("<App/>", () => {
  it("renders", () => {
    const { asFragment } = renderWithRedux(reducer, {}, <App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
