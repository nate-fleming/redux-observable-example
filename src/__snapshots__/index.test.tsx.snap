// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<McgContainer/> renders 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="Table"
        src="table.png"
      />
      <button
        class="Start-Button"
      >
        Start
      </button>
      <div
        style="display: flex; width: 100%;"
      />
    </header>
  </div>
</DocumentFragment>
`;
